### ISSUE TITLE
Unexpected WriteEvent parameter value in netperf file

### ISSUE THREAD

**Issue by jiangzeng01** [dotnet/coreclr#24724]
Test:
coreclr\tests\src\tracing\eventsource\eventsourcetrace\EventSourceTrace.cs

Repro steps:
1. Change the line 23 of EventSourceTrace.cs to:
_internal void MathResult(int x, int y, int z, string formula) { this.WriteEvent(1, x, y, z, x, y, z, formula); }_
2. Build & run the test
3. Check MathResults events in the netperf file generated, the parameter **formula** is showing some strange values which are not expected.

Other concern:
With this change, I passed x,y,z as parameters to WriteEvent twice, should it be showing twice in the netperf file? For now, it shows only once.

![image](https://user-images.githubusercontent.com/49379637/58194427-12c2ca00-7cf8-11e9-8b05-eea292d4c6f5.png)

**Comment by hoyosjs** [comment]
By design. This falls to:
```c#
protected unsafe void WriteEvent(int eventId, params object[] args)
```
If a debugger is attached you'll get a argument mismatch log in the Debugger from TraceEvent. 

However, this seems like a really easy way to get screwed up events without even realizing. To make it worse, the message provided is near to useless: `The parameters to the Event method do not match the parameters to the WriteEvent method. This may cause the event to be displayed incorrectly.` We have no information on the EventId or the EventSource. We might need a way to better diagnose the mismatch here.

@dotnet/dotnet-diag

**Comment by sywhang** [comment]
> With this change, I passed x,y,z as parameters to WriteEvent twice, should it be showing twice in the netperf file? For now, it shows only once.

You passed it once, with an incorrect number of parameters. 

Improving the diagnostic message passed to the debugger can certainly be done but it doesn't seem that high of a priority now. I'll file a separate issue for that.
